mozdownload/scraper.py

```diff
@@ -274,15 +279,16 @@ def total_seconds(td):
 
         attempt = 0
 
-        if not os.path.isdir(self.directory):
-            os.makedirs(self.directory)
-
         # Don't re-download the file
         if os.path.isfile(os.path.abspath(self.target)):
             self.logger.info("File has already been downloaded: %s" %
                              (self.target))
             return
 
+        directory = os.path.dirname(self.target)
+        if directory and not os.path.isdir(directory):
```

Comment:
We always have a valid directory here. It should never be `None`. So you can get rid of the first check.